pros/serial/interactive/UploadProjectModal.py

```diff
@@ -126,11 +126,20 @@ def confirm(self, *args, **kwargs):
         from pros.cli.upload import upload
         from click import get_current_context
         kwargs = {'path': None, 'project': self.project, 'port': self.port.value}
+        savable_kwargs = {}
         if self.project.target == 'v5':
-            kwargs['remote_name'] = self.advanced_options['name'].value
-            kwargs['slot'] = int(self.advanced_options['slot'].value[0])  # XXX: the first character is the slot number
-            kwargs['description'] = self.advanced_options['description'].value
-            kwargs['compress_bin'] = self.advanced_options['compress_bin'].value
+            savable_kwargs['remote_name'] = self.advanced_options['name'].value
+            # XXX: the first character is the slot number
+            savable_kwargs['slot'] = int(self.advanced_options['slot'].value[0])
+            savable_kwargs['description'] = self.advanced_options['description'].value
+            savable_kwargs['compress_bin'] = self.advanced_options['compress_bin'].value
+
+        if self.save_settings.value:
+            self.project.project_name = savable_kwargs['remote_name']
```

Comment:
Why is this different?

Reply:
because there's no key called `remote_name` on `Project` as far as I could tell. all the other functions seem to use `project.name` which is an `@property` that pulls from either `self.project_name` or creates a name based on some available information

Reply:
also project name isn't an upload option

Reply:
Ah, I think it's because we're not setting the default name correctly on L107. I think it would be `self.project.upload_options.get('remote_name', self.project.name)`